internal/fluentbit/config/builder/config_builder.go

```diff
@@ -1,129 +1,113 @@
+	client.Reader
+	BuilderConfig
 }
 
-func createLuaDedotFilter(logPipeline *telemetryv1alpha1.LogPipeline) string {
-	output := logPipeline.Spec.Output
-	if !logpipelineutils.IsHTTPDefined(&output) || !output.HTTP.Dedot {
-		return ""
+func (b *ConfigBuilder) Build(ctx context.Context, allPipelines []telemetryv1alpha1.LogPipeline) (*FluentBitConfig, error) {
+	config := FluentBitConfig{
+		SectionsConfig:  make(map[string]string),
+		FilesConfig:     make(map[string]string),
+		EnvConfigSecret: make(map[string][]byte),
+		TLSConfigSecret: make(map[string][]byte),
 	}
 
-	return NewFilterSectionBuilder().
-		AddConfigParam("name", "lua").
-		AddConfigParam("match", fmt.Sprintf("%s.*", logPipeline.Name)).
-		AddConfigParam("script", "/fluent-bit/scripts/filter-script.lua").
-		AddConfigParam("call", "kubernetes_map_keys").
-		Build()
-}
+	for _, pipeline := range allPipelines {
+		if !isLogPipelineReconcilable(allPipelines, &pipeline) || !pipeline.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		sectionsConfigMapKey := pipeline.Name + ".conf"
 
-func validateCustomSections(pipeline *telemetryv1alpha1.LogPipeline) error {
-	customOutput := pipeline.Spec.Output.Custom
-	if customOutput != "" {
-		_, err := config.ParseCustomSection(customOutput)
+		sectionsConfigMapContent, err := BuildFluentBitSectionsConfig(&pipeline, b.BuilderConfig)
 		if err != nil {
-			return err
+			return nil, fmt.Errorf("unable to build section: %w", err)
 		}
-	}
 
-	for _, filter := range pipeline.Spec.Filters {
-		_, err := config.ParseCustomSection(filter.Custom)
+		filesConfig := BuildFluentBitFilesConfig(&pipeline)
+
+		envConfigSecret, err := b.BuildEnvConfigSecret(ctx, allPipelines)
 		if err != nil {
-			return err
+			return nil, fmt.Errorf("unable to build env config: %w", err)
 		}
-	}
 
-	return nil
-}
+		tlsConfigSecret, err := b.BuildTLSFileConfigSecret(ctx, allPipelines)
+		if err != nil {
+			return nil, fmt.Errorf("unable to build tls secret: %w", err)
+		}
 
-func validateOutput(pipeline *telemetryv1alpha1.LogPipeline) error {
-	if !logpipelineutils.IsAnyDefined(&pipeline.Spec.Output) {
-		return fmt.Errorf("%w: No output plugin defined", ErrInvalidPipelineDefinition)
+		config.addSections(map[string]string{sectionsConfigMapKey: sectionsConfigMapContent})
+		config.addFiles(filesConfig)
+		config.addEnvConfigSecret(envConfigSecret)
+		config.addTLSConfigSecret(tlsConfigSecret)
 	}
 
-	return nil
+	return &config, nil
 }
 
-func validateInput(pipeline *telemetryv1alpha1.LogPipeline) error {
-	if pipeline.Spec.Input.OTLP != nil {
-		return fmt.Errorf("%w: cannot use OTLP input for pipeline in FluentBit mode", ErrInvalidPipelineDefinition)
-	}
-
-	if pipeline.Spec.Input.Application == nil {
-		return nil
+// isLogPipelineReconcilable checks if logpipeline is ready to be rendered into the fluentbit configuration.
+// A pipeline is reconcilable if it is not being deleted, all secret references exist, and is not above the pipeline limit.
```

Comment:
This documentation comment doesn't match with what the function is actually checking

Reply:
And actually do we even need this check here?
We anyway only call the `Build` func for reconilable pipelines